### ISSUE TITLE
[Bug]: [ResourceGroup]Querynode crashed when calling transfer_node and transfer_replica

### ISSUE THREAD

**Issue by yanliang567** [milvus-io/milvus#22363]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.2.0-20230221-eac4523c
- Deployment mode(standalone or cluster): cluster with 3 query nodes
- MQ type(rocksmq, pulsar or kafka):   pulsar
```

### Current Behavior

querynode crashed

### Expected Behavior

nothing crashed

### Steps To Reproduce

```markdown
1. create 2 collections: c1 and c2, load c1 and c2 with 1 replica into default rg(2 nodes)
2. create 2 rgs: rgA(1 node) and rgB(0 node)
3. transfer 1 node from default rg to rgB and transfer replica of c2 from default to rgB
4. transfer replica of c1 from default to rgB
one of querynode crashed
```

### Milvus Log

crashed around 12:00 2-23
```
yanliang-cluster-1cu-milvus-datanode-6cf8684b97-kgps9             1/1     Running     0               45h
yanliang-cluster-1cu-milvus-indexnode-5d7b498b6b-7tkr9            1/1     Running     0               45h
yanliang-cluster-1cu-milvus-mixcoord-596b4cc95-nvvd6              1/1     Running     0               45h
yanliang-cluster-1cu-milvus-proxy-59cb4c745-jfxqz                 1/1     Running     0               45h
yanliang-cluster-1cu-milvus-querynode-79f8f8877c-bzmcm            1/1     Running     1 (154m ago)    45h
yanliang-cluster-1cu-milvus-querynode-79f8f8877c-qgl28            1/1     Running     0               45h
yanliang-cluster-1cu-milvus-querynode-79f8f8877c-rcw79            1/1     Running     0               45h
```
```
[2023/02/23 03:59:07.602 +00:00] [DEBUG] [querynode/task_read.go:81] [tr/searchTask(3323c6d1a27ebbaa)] [msg="start to process"] [duration=9.466413ms]
[2023/02/23 03:59:07.602 +00:00] [DEBUG] [querynode/task_read.go:81] [tr/searchTask(503f5817330281dc)] [msg="start to process"] [duration=8.794622ms]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x2bb74f0]

goroutine 2449122360 [running]:
github.com/milvus-io/milvus/internal/querynode.(*distribution).GetCurrent(0x0, {0x0?, 0xc00080f1e0?, 0xc003199408?})
        /go/src/github.com/milvus-io/milvus/internal/querynode/distribution.go:98 +0x50
github.com/milvus-io/milvus/internal/querynode.(*ShardCluster).GetSegmentInfos(0xc0035700c0)
        /go/src/github.com/milvus-io/milvus/internal/querynode/shard_cluster.go:866 +0x36
github.com/milvus-io/milvus/internal/querynode.(*QueryNode).GetDataDistribution(0xc0008fe1e0, {0x3c4b380?, 0x3d1b1e0?}, 0xc0034d2150)
        /go/src/github.com/milvus-io/milvus/internal/querynode/impl.go:1348 +0xd92
github.com/milvus-io/milvus/internal/distributed/querynode.(*Server).GetDataDistribution(0xf?, {0x41ecd58?, 0xc0034d2270?}, 0x10?)
        /go/src/github.com/milvus-io/milvus/internal/distributed/querynode/service.go:337 +0x2b
github.com/milvus-io/milvus/internal/proto/querypb._QueryNode_GetDataDistribution_Handler.func1({0x41ecd58, 0xc0034d2270}, {0x3c3aca0?, 0xc0034d2150})
        /go/src/github.com/milvus-io/milvus/internal/proto/querypb/query_coord.pb.go:5811 +0x78
github.com/milvus-io/milvus/internal/util/logutil.UnaryTraceLoggerInterceptor({0x41ecd58?, 0xc0034d21e0?}, {0x3c3aca0, 0xc0034d2150}, 0x41d71c0?, 0xc003492f60)
        /go/src/github.com/milvus-io/milvus/internal/util/logutil/grpc_interceptor.go:22 +0x49
```

**Comment by weiliu1031** [comment]
same as https://github.com/milvus-io/milvus/issues/22312
and fix on #22317 

please verify on this

**Comment by yanliang567** [comment]
verified on 2.2.0-20230228-91d251ab
